staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store.go

```diff
@@ -966,6 +966,15 @@ func (e *Store) updateForGracefulDeletionAndFinalizers(ctx context.Context, name
 			if err != nil {
 				return nil, err
 			}
+			if existingAnnotations := existingAccessor.GetAnnotations(); existingAnnotations != nil {
```

Comment:
no need for the check. In go you can access a nil map without panic.

Reply:
oh, right, writing to a `nil` map panic but not reading.